scale/storage/models.py

```diff
@@ -217,6 +217,52 @@ def download_files(self, file_downloads):
             wp_file_downloads = wp_dict[wp_id][1]
             workspace.download_files(wp_file_downloads)
 
+    def filter_files(self, started=None, ended=None, time_field=None, file_name=None):
+        """Returns a query for Scale files that is filtered on the given fields.
+
+        :param started: Query Scale files updated after this amount of time.
+        :type started: :class:`datetime.datetime`
+        :param ended: Query Scale files updated before this amount of time.
+        :type ended: :class:`datetime.datetime`
+        :keyword time_field: The time field to use for filtering.
+        :type time_field: string
+        :param file_name: Query Scale files with the given file name.
+        :type file_name: str
+        :returns: The Scale file query
+        :rtype: :class:`django.db.models.QuerySet`
+        """
+
+        # Fetch a list of source files
+        files = ScaleFile.objects.all()
+        files = files.select_related('workspace')
+        files = files.defer('workspace__json_config')
+        files = files.prefetch_related('countries')
+
+        # Apply file_name filtering if present
+        if file_name:
+            files = files.filter(file_name=file_name)
+            files = files.order_by('last_modified')
+            return files
```

Comment:
File name could match more than one file, so don't stop short here. Continue applying the rest of the filtering.